### PR TITLE
Action Sheet Component

### PR DESCRIPTION
**Pull request by StefanGreve**
Part 2 of 2 for resolving #12.

### TODO
 
 - [x] Configure unit tests (fix `Error: can't resolve src/styles.scss in ...`)
 - [x] implement `ActionSheetComponent`
 - [x] use directives to refactor inputs
 - [x] add `checked` directive

### CODE REVIEW

**Comment by StefanGreve** [comment]
Idea: change implementation of `ThemeSwitcher` such that global CSS variables are written to a `style` element beneath the `body` tag?

File: src/app/app.component.ts [hunk 1]
```diff
@@ -31,71 +34,54 @@ export class AppComponent implements AfterViewInit, OnInit, OnDestroy {
   ];
 
   @ViewChild(AlertComponent, {static: false})
-  alert: AlertComponent | undefined;
-
-  alertSubject$: BehaviorSubject<boolean> = new BehaviorSubject(false);
```
Comment (StefanGreve) [inline src/app/app.component.ts:36]: For the time being, I removed Subject and Subscription code. If, for some reason, one would like to keep track of the opened/closed state of an overlay, using `signals` would be much better here, anyway.

File: src/app/components/alert/alert.component.ts [hunk 2]
```diff
@@ -9,45 +10,49 @@ import { v4 as uuid } from "uuid";
...
+  hostDirectives: [
+    {
+      directive: TitleDirective,
+      inputs: ["title"],
+    },
+  ]
 })
 export class AlertComponent implements OnInit {
+  // dependency injection
+  private titleDirective = inject(TitleDirective);
```
Comment (StefanGreve) [inline src/app/components/alert/alert.component.ts:23]: Moving forward, we will use this style of dependency injection everywhere, which renders the use of empty constructors as useless.

File: src/app/app.component.html [hunk 3]
```diff
@@ -23,5 +23,7 @@ <h2>Settings</h2>
     <h1 class="title">Hello, World!</h1>
     <adv-code-line code="test" title="GPG Key" />
   </section>
-  <adv-alert content="Hello, World!" [actions]=testAlert>Alert!</adv-alert>
+  <adv-alert title="Test Alert" [hidden]=false content="Hello, World!" [actions]=alertActions>Alert!</adv-alert>
+  <button (click)=openActionSheet()>ActionSheet</button>
+  <adv-action-sheet [actions]=actionSheetActions></adv-action-sheet>
```
Comment (KidBased) [inline src/app/app.component.html:28]: I always find it quite irritating that you don't add any quotation marks around property bindings or methods. It makes the property or event look like a normal string in my code editor due to the syntax highlighting. 

```diff
- <button (click)=openActionSheet()>ActionSheet</button>
+ <button (click)="openActionSheet()">ActionSheet</button>
```

File: src/app/components/action-sheet/action-sheet.component.html
Comment (KidBased) [inline src/app/components/action-sheet/action-sheet.component.html:1]: We can use the new overlay component I created here. It's better when the overlay functionality is decoupled from the actual component.

File: src/app/directives/inputs/checked.directive.ts
Comment (KidBased) [inline src/app/directives/inputs/checked.directive.ts:1]: Personally, I would not have created a directive for something like that, seems a bit overkill to me. I think it's better to handle these kind of things inside the component template.

Comment (KidBased) [reply src/app/directives/inputs/checked.directive.ts:1]: The general idea of what you're doing is still pretty neat though.

**Review by KidBased — APPROVED** [review]
Looks good.

**Comment by KidBased** [comment]
> * [x]  Idea: change implementation of `ThemeSwitcher` such that global CSS variables are written to a `style` element beneath the `body` tag?
> 
> Answer: This is not possible without retendering the view:
> 
> ```ts
> private createTheme(themeDefinition: ThemeDefinition): void {
>     const body = document.body;
> 
>     if (body.firstChild?.nodeName === "STYLE") body.removeChild(body.firstChild);
> 
>     const style = document.createElement("style");
> 
>     const variables = Object.keys(themeDefinition).map(key => {
>       // Custom CSS properties (variables) require a double hyphen as leading
>       // characters. By convention, TypeScript properties use camelCase which
>       // we also replace here with a single hyphen character as a word separator
>       const variable = `--${key.split(/(?=[A-Z])/).join("-").toLowerCase()}`;
>       const value = themeDefinition[key];
>       return `\t${variable}: "${value}";`;
>     }).join("\n");
> 
>     // NOTE: it would be necessary to mark this function for the change detection,
>     // but there is no trivial way to substantially modify the DOM from within a service.
>     // Any attempts of passing ChangeDetectorRef caused an NullInjectionError.
>     style.textContent = `:root {\n${variables}\n}`;
>     body.insertBefore(style, body.firstChild);
>   }
> ```
> 
> So any modifications to the DOM would need to work with already existing elements.

We should discuss that more in detail because I'm not sure if I understood that correctly.

**Pull request merged**